tools/d2g/indexeddockerimage.go

```diff
@@ -1,59 +1,31 @@
 package d2g
 
 import (
+	"encoding/json"
 	"fmt"
 	"path/filepath"
-	"strings"
 
-	"github.com/taskcluster/shell"
 	"github.com/taskcluster/taskcluster/v55/tools/d2g/genericworker"
 )
 
-func (idi *IndexedDockerImage) PrepareCommands() []string {
-	findArtifactURL := fmt.Sprintf("${TASKCLUSTER_PROXY_URL}/index/v1/task/%s/artifacts/%s", idi.Namespace, idi.Path)
-	filename := filepath.Base(idi.Path)
-	commands := []string{
-		fmt.Sprintf(`curl -fsSL -o %s "%s"`, filename, findArtifactURL),
-	}
-
-	handleFileExtentions(filename, &commands)
-
-	return commands
-}
-
 func (idi *IndexedDockerImage) FileMounts() ([]genericworker.FileMount, error) {
-	return []genericworker.FileMount{}, nil
+	indexedContent := genericworker.IndexedContent{
+		Artifact:  idi.Path,
+		Namespace: idi.Namespace,
+	}
+	raw, err := json.MarshalIndent(&indexedContent, "", "  ")
+	if err != nil {
+		return nil, fmt.Errorf("cannot marshal indexed content %#v into json: %w", indexedContent, err)
+	}
+	return []genericworker.FileMount{
+		{
+			Content: json.RawMessage(raw),
+			// TODO check if this could conflict with other files(?)
+			File: filepath.Base(idi.Path),
```

Comment:
```suggestion
			File: "dockerimage",
```

Note, below I propose supporting lz4 compression in mounted Read Only Directories and mounted Writable Directory Caches so that the mounts feature automatically performs the lz4 decompression. If we follow this approach, here we would need to check if this is a `.lz4` file, and if so, set `Directory` instead of `File`, and set `Format` to `"lz4"`. This would cause the mounts feature to decompress the file. Presumably the lz4 decompression simply strips the file `.lz4` filename suffix, and the compressed file does not have the ability to choose the target file name.

We should probably do a review of the compression formats we support, to see which are pure compression, and which are a combination of archiving and compression. I believe lz4 is pure compression, and no archiving. Pure compression probably makes sense also for file artifacts, whereas archive formats probably only make sense for mounted directories. Currently I think we don't support any compression formats for file artifacts.